--- dwm.c	2025-03-08 00:32:09.800363409 +0100
+++ dwm_patched.c	2025-03-08 00:37:10.502978565 +0100
@@ -1686,29 +1686,31 @@
 void
 tile(Monitor *m)
 {
-	unsigned int i, n, h, mw, my, ty;
-	Client *c;
+    unsigned int i, n, w, mh, mx, tx;
+    Client *c;
 
-	for (n = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), n++);
-	if (n == 0)
-		return;
+    for (n = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), n++);
+    if (n == 0)
+        return;
 
-	if (n > m->nmaster)
-		mw = m->nmaster ? m->ww * m->mfact : 0;
-	else
-		mw = m->ww;
-	for (i = my = ty = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++)
-		if (i < m->nmaster) {
-			h = (m->wh - my) / (MIN(n, m->nmaster) - i);
-			resize(c, m->wx, m->wy + my, mw - (2*c->bw), h - (2*c->bw), 0);
-			if (my + HEIGHT(c) < m->wh)
-				my += HEIGHT(c);
-		} else {
-			h = (m->wh - ty) / (n - i);
-			resize(c, m->wx + mw, m->wy + ty, m->ww - mw - (2*c->bw), h - (2*c->bw), 0);
-			if (ty + HEIGHT(c) < m->wh)
-				ty += HEIGHT(c);
-		}
+    if (n > m->nmaster)
+        mh = m->nmaster ? m->wh * m->mfact : 0;
+    else
+        mh = m->wh;
+    
+    for (i = mx = tx = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++) {
+        if (i < m->nmaster) {
+            w = (m->ww - mx) / (MIN(n, m->nmaster) - i);
+            resize(c, m->wx + mx, m->wy, w - (2*c->bw), mh - (2*c->bw), 0);
+            if (mx + WIDTH(c) < m->ww)
+                mx += WIDTH(c);
+        } else {
+            w = (m->ww - tx) / (n - i);
+            resize(c, m->wx + tx, m->wy + mh, w - (2*c->bw), m->wh - mh - (2*c->bw), 0);
+            if (tx + WIDTH(c) < m->ww)
+                tx += WIDTH(c);
+        }
+    }
 }
 
 void
